### ISSUE TITLE
`no-incorrect-template-string-interpolation`: the rule seems overly strict right now

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3312]
### Description

The issue title here is a bit vague, but with it I attempted to highlight a couple of things I don't currently like about the rule.

First, I doubt it should be in the `unopinionated` preset. Technically, there is nothing "incorrect" in the examples the rule marks as invalid. Obviously most of the time these are mistakes, especially in the second example (`Hello $user.name}`), but I think they can't be universally and confidently treated as something wrong, hence shouldn't be put in that preset.

Second, the first example, `Hello {name}`, fairly easily yields a false positive:

```ts
const someGeneratedCode = `
import {foo} from 'bar';
...
`;
```

I don't know of a perfect fix for these kinds of situations but here are some proposed one:
- Add the ability to disable certain kinds of potential problems: it seems like rule addition was initially [motivated by JSX->template strings migration](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2150), which is itself quite a narrow case.
- And/or add hardcoded likely valid examples like JS `import`/`export` statements. That might cause false negatives.

### Examples

Example given above
